### ISSUE TITLE
`pyhmmer.nhmmer` throws  error when the number of FASTA seqs is larger than 64

### ISSUE THREAD

**Issue by moshi4** [althonos/pyhmmer#62]
Hi @althonos, this is a bug report about `pyhmmer.nhmmer`.

I encountered an error when entering a multi-fasta in pyhmmer.nhmmer with [pybarrnap](https://github.com/moshi4/pybarrnap). 
After investigating the conditions for reproducing the error, it appears that the error occurs when the number of FASTA seqs is greater than 64. Also, the error contents are not always the same, such as `Segmentation fault`, `corrupted size vs. prev_size`, `munmap_chunk(): invalid pointer`.

You can reproduce the error with the following examples data and commands.

[examples.zip](https://github.com/althonos/pyhmmer/files/14560785/examples.zip)

```shell
pybarrnap examples/64seqs_success.fa
pybarrnap examples/65seqs_failed.fa
pybarrnap examples/91seqs_failed.fa
```

**Comment by althonos** [comment]
Thanks for the report @moshi4, 

There was a bug in the way i was computing array reallocation which caused arrays of 64 items to never be reallocated to a larger capacity, which was then causing some out-of-buffer writes and causing various memory issues later.

I fixed the problem, confirmed `pybarrnap` works properly now, and will push a patch.

**Comment by althonos** [comment]
Fixed in `v0.10.9` :+1:
